### PR TITLE
Fix ClipTimeToViewportTimeRangePercentFormatProvider Get Viewport Duration

### LINKED ISSUE #16: ClipTimeToViewportTimeRangePercentFormatProvider's getViewPortDuration is always 0

### PR DESCRIPTION
**Pull request by trgibeau**
Closes #16 by updating clone to keep the viewport container and custom formatter instance instead of creating a new one

### CODE REVIEW

File: src/Microsoft.Performance.SDK/Processing/Projection/ClipTimeToViewportProjection.cs [hunk 1]
```diff
@@ -158,20 +158,20 @@ private struct ClipTimeToViewportTimeRangeColumnGenerator<TGenerator>
             {
                 internal TGenerator Generator { get; }
 
-                internal VisibleTableRegionContainer ViewPortContainer { get; }
+                private readonly VisibleTableRegionContainer viewportContainer;
 
                 public ClipTimeToViewportTimeRangeColumnGenerator(TGenerator timestampGenerator)
```
Comment (jmaxson-ms) [inline src/Microsoft.Performance.SDK/Processing/Projection/ClipTimeToViewportProjection.cs:163]: >timestampGenerator [](start = 77, length = 18)

not part of this PR, but if you have any other changes to make, please consider renaming this to timeRangeGenerator :)

**Review by jmaxson-ms — DISMISSED** [review]
:shipit:

File: src/Microsoft.Performance.SDK/Processing/Projection/ClipTimeToViewportProjection.cs [hunk 2]
```diff
@@ -255,26 +259,29 @@ public object GetFormat(Type formatType)
                 public object Clone()
                 {
                     var result = new ClipTimeToViewportTimeRangePercentColumnGenerator<TGenerator>(
-                        ViewportSensitiveProjection.CloneIfViewportSensitive(this.timeRangeColumnGenerator.Generator));
+                        ViewportSensitiveProjection.CloneIfViewportSensitive(this.timeRangeColumnGenerator.Generator),
+                        this.viewportContainer,
```
Comment (mslukebo) [inline src/Microsoft.Performance.SDK/Processing/Projection/ClipTimeToViewportProjection.cs:263]: VisibleTableRegionContainer is mutable (VisibleTableRegion has a setter).  Perhaps make a new instance and set its property for the cloned instance

**Review by zachnew-msft — APPROVED** [review]

File: src/Microsoft.Performance.SDK/Processing/Projection/ClipTimeToViewportProjection.cs (hunk 2 above)
Comment (mslukebo) [reply src/Microsoft.Performance.SDK/Processing/Projection/ClipTimeToViewportProjection.cs:263]: Resolved offline. The container needs to be shared with the cloned instance.

**Review by jmaxson-ms — APPROVED** [review]

**Review by mslukebo — APPROVED** [review]
